### ISSUE TITLE
<typeinfo> must be included before typeid is used

### ISSUE THREAD

**Issue by hypermagik** [microsoft/vscode-cpptools#1067]
The following piece of code shows this error, even though typeinfo is included:
```C++
#include <typeinfo>

int main() {
    const std::type_info &ti = typeid(main);
    return 0;
}
```
```
file: 'file:///.../test.cpp'
severity: 'Error'
message: '<typeinfo> must be included before typeid is used'
at: '4,32'
source: ''
```

Operating System and version: _Ubuntu 16.04.3 LTS_.

VS Code version and if you are using the Insiders build: _1.16.1_ (not Insiders build).

C/C++ extension version: _0.13.0_.

Other extensions you installed and if the issue persists after disabling them: _issue persists after disabling all other extensions_.

Relevant settings from your settings.json, c_cpp_properties.json, and/or launch.json files:
_"C_Cpp.intelliSenseEngine": "Default"_.

**Comment by sean-mcmanus** [comment]
Thanks for reporting it. It repros on Linux, but not Windows.

**Comment by ortix** [comment]
This is also broken on macOS Sierra

**Comment by EECSGEEK** [comment]
<h3><b>This is also broken on Win-Platform</b></h3>
<img src="http://p3.cdn.img9.top/ipfs/QmTeHsD1MLwVCodTFWrPvJkxvc4KAapgnpLu6mJRFCnWGF?3.png" />

**Comment by sean-mcmanus** [comment]
@EECSGEEK The error shouldn't occur on Windows unless the typeinfo header (or a header it includes) fails to be located (or fails to get parsed). Are there other errors in your problems window?

**Comment by EECSGEEK** [comment]
@sean-mcmanus 
<h3>Program one(g++include cunistd.h)
<br />(g++ compile is ok)
</h3>
<img src="http://p0.cdn.img9.top/ipfs/QmaXwcThp2wkzeEYodHK9tG1AKNiSP7dL95bebMSsYBB9E?0.PNG" alt="" />
<h3>Program 2
<br />But vscode [error]typeinfo must be included before typeid is used</h3>
<img src="http://p2.cdn.img9.top/ipfs/QmbDHZLDWqumobjU3nqE5cUgaYFREgZzgCmLRsLkJi8sKV?2.png" alt="" />
<h3>My setting</h3>
<img src="http://p4.cdn.img9.top/ipfs/Qmf71Wyn8ZzoK8UtYfCGKJ9M6gkZhWEKrxbDg5ujyaBsfZ?4.PNG" alt="" />

**Comment by sean-mcmanus** [comment]
@EECSGEEK Oh, yeah, you're using mingw with clang-x64 mode -- that makes sense why it would behave like on Linux. The issue doesn't repro with the typeinfo header defined in the Windows SDK.

**Comment by sean-mcmanus** [comment]
I've verified the fix is in our pending update branch for February.

**Comment by sean-mcmanus** [comment]
Fixed with the preview .vsix at https://github.com/Microsoft/vscode-cpptools/releases/tag/v0.15.0-insiders (our final release is planned for next week).

**Comment by sean-mcmanus** [comment]
@HypermagiK @Ortix92 @EECSGEEK The bogus `<typeinfo> must be included before typeid is used` error has been fixed with 0.15.0. We're still aware of additional bogus errors that we have issues tracking that we plan to have fixes in the next couple months or so. If you notice a new issue, providing a repro will help us get a fix sooner. Thanks.
